packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1360 @@
+        cellContentIndex: number,
+        columnIndex?: number,
+        rowIndex?: number
+    ): boolean {
+        if (!rowElements) {
+            return false;
+        }
+        const newColumnIndex = columnIndex ?? this.columnIndex;
+        const newRowIndex = rowIndex ?? this.rowIndex;
+
+        if (
+            newColumnIndex >= 0
+            && newColumnIndex < rowElements.cells.length
+            && cellContentIndex >= 0
+            && cellContentIndex
+                < rowElements.cells[newColumnIndex]!.cell.cellView
+                    .tabbableChildren.length
+        ) {
+            this.setCellContentFocusState(
+                cellContentIndex,
+                newRowIndex,
+                newColumnIndex,
+                true
+            );
+            return true;
+        }
+
+        return false;
+    }
+
+    private trySetCellActionMenuFocus(
+        rowElements: TableRowFocusableElements,
+        columnIndex?: number,
+        rowIndex?: number
+    ): boolean {
+        const newColumnIndex = columnIndex ?? this.columnIndex;
+        const newRowIndex = rowIndex ?? this.rowIndex;
+
+        if (
+            newColumnIndex >= 0
+            && newColumnIndex < rowElements.cells.length
+            && rowElements.cells[newColumnIndex]!.actionMenuButton
+        ) {
+            this.setCellActionMenuFocusState(newRowIndex, newColumnIndex, true);
+            return true;
+        }
+
+        return false;
+    }
+
+    private setCellActionMenuFocusState(
+        rowIndex: number,
+        columnIndex: number,
+        focusElement: boolean
+    ): void {
+        this.focusType = TableFocusType.cellActionMenu;
+        this.setRowCellFocusState(columnIndex, rowIndex, focusElement);
+    }
+
+    private setCellContentFocusState(
+        cellContentIndex: number,
+        rowIndex: number,
+        columnIndex: number,
+        focusElement: boolean
+    ): void {
+        this.focusType = TableFocusType.cellContent;
+        this.cellContentIndex = cellContentIndex;
+        this.setRowCellFocusState(columnIndex, rowIndex, focusElement);
+    }
+
+    private setRowFocusState(rowIndex?: number): void {
+        this.focusType = TableFocusType.row;
+        if (rowIndex !== undefined) {
+            this.rowIndex = rowIndex;
+        }
+    }
+
+    private setCellFocusState(
+        columnIndex: number,
+        rowIndex: number,
+        focusElement: boolean
+    ): void {
+        this.focusType = TableFocusType.cell;
+        this.setRowCellFocusState(columnIndex, rowIndex, focusElement);
+    }
+
+    private setRowCellFocusState(
+        columnIndex: number,
+        rowIndex: number,
+        focusElement: boolean
+    ): void {
+        this.rowIndex = rowIndex;
+        this.columnIndex = columnIndex;
+        if (focusElement) {
+            this.focusCurrentRow(true);
+        }
+    }
+
+    private isVisualRow(row: unknown): row is TableRow | TableGroupRow {
```

Comment:
think we went for the term resolved right? Should this be `isResolvedRow` or `isResolvedRowType`?

Reply:
Updated